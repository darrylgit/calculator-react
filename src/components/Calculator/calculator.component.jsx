import React, { useState } from 'react';

import Display from '../Display/display.component';
import Keypad from '../Keypad/keypad.component';

import { CalculatorContainer } from './calculator.styles';
import CalcContext from '../../contexts/calcContext';
import newValueHandler from '../../logic/newValueHandler/newValueHandler';

const Calculator = () => {
  const [values, setValues] = useState([]);
  const pushValue = inputValue =>
    setValues(newValueHandler(values, inputValue));

  return (
    <CalculatorContainer>
      <CalcContext.Provider
        value={{
          values,
          pushValue
        }}
      >
        <Display />
        <Keypad />
      </CalcContext.Provider>
    </CalculatorContainer>
  );
};

export default Calculator;
