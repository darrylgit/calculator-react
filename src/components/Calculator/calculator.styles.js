import styled from 'styled-components';

export const CalculatorContainer = styled.div`
  width: 36rem;
  height: 64rem;
  border: 2px solid #6fa7c6;
  background-color: white;
  border-radius: 20px;
  padding: 0;
  -webkit-transform: scale(1);
  overflow: hidden;
  background-repeat: no-repeat;
  padding-bottom: 0.5rem;

  @media only screen and (max-width: 26.55em) {
    height: 100%;
    width: 100%;
    margin-top: 0;
    border-radius: 0;
  }

  @media only screen and (max-height: 40em) {
    height: 100%;
  }

  // Mobile landscape mode
  @media screen and (min-device-aspect-ratio: 1/1) and (orientation: landscape) and (hover: none) and (max-height: 40em) {
    width: 100vw;
    border-radius: 0;
  }
`;
