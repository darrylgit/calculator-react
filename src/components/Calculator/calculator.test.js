import { mount } from 'enzyme';
import React from 'react';
import Calculator from './calculator.component';
import { faItalic } from '@fortawesome/free-solid-svg-icons';

it('renders without crashing', () => {
  let test = mount(<Calculator />);
  expect(test.debug()).toMatchSnapshot();
});
