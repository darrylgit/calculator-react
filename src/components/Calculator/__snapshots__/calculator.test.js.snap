// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
"<Calculator>
  <styled.div>
    <div className=\\"sc-AxgMl fWFYAo\\">
      <Display>
        <styled.div>
          <div className=\\"sc-AxjAm ePhIG\\">
            <div className=\\"outputs\\">
              <div className=\\"preview\\" />
              <div className=\\"main\\">
                 
              </div>
            </div>
          </div>
        </styled.div>
      </Display>
      <KeyPad>
        <styled.div>
          <div className=\\"sc-AxhUy IYOeL\\">
            <InputKey value=\\"CLEAR\\" label=\\"C\\" type=\\"clear\\">
              <_default type=\\"clear\\" label=\\"C\\" value=\\"CLEAR\\" onClick={[Function: onClick]}>
                <styled.div>
                  <div className=\\"sc-AxhCb jdovTQ\\">
                    <styled.div>
                      <div className=\\"sc-AxirZ eFNAXz\\">
                        <styled.div type=\\"clear\\" label=\\"C\\" value=\\"CLEAR\\" onClick={[Function: onClick]}>
                          <div type=\\"clear\\" label=\\"C\\" value=\\"CLEAR\\" onClick={[Function: onClick]} className=\\"sc-AxiKw fcXlqn\\">
                            C
                          </div>
                        </styled.div>
                      </div>
                    </styled.div>
                  </div>
                </styled.div>
              </_default>
            </InputKey>
            <InputKey value=\\"BACKSPACE\\" label={{...}} type=\\"util\\">
              <_default type=\\"util\\" label={{...}} value=\\"BACKSPACE\\" onClick={[Function: onClick]}>
                <styled.div>
                  <div className=\\"sc-AxhCb jdovTQ\\">
                    <styled.div>
                      <div className=\\"sc-AxirZ eFNAXz\\">
                        <styled.div type=\\"util\\" label={{...}} value=\\"BACKSPACE\\" onClick={[Function: onClick]}>
                          <div type=\\"util\\" label={{...}} value=\\"BACKSPACE\\" onClick={[Function: onClick]} className=\\"sc-AxiKw kocETC\\">
                            <FontAwesomeIcon icon={{...}} border={false} className=\\"\\" mask={{...}} fixedWidth={false} inverse={false} flip={{...}} listItem={false} pull={{...}} pulse={false} rotation={{...}} size={{...}} spin={false} symbol={false} title=\\"\\" transform={{...}} swapOpacity={false}>
                              <svg aria-hidden=\\"true\\" focusable=\\"false\\" data-prefix=\\"fas\\" data-icon=\\"arrow-left\\" className=\\"svg-inline--fa fa-arrow-left fa-w-14 \\" role=\\"img\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 448 512\\" style={{...}}>
                                <path fill=\\"currentColor\\" d=\\"M257.5 445.1l-22.2 22.2c-9.4 9.4-24.6 9.4-33.9 0L7 273c-9.4-9.4-9.4-24.6 0-33.9L201.4 44.7c9.4-9.4 24.6-9.4 33.9 0l22.2 22.2c9.5 9.5 9.3 25-.4 34.3L136.6 216H424c13.3 0 24 10.7 24 24v32c0 13.3-10.7 24-24 24H136.6l120.5 114.8c9.8 9.3 10 24.8.4 34.3z\\" style={{...}} />
                              </svg>
                            </FontAwesomeIcon>
                          </div>
                        </styled.div>
                      </div>
                    </styled.div>
                  </div>
                </styled.div>
              </_default>
            </InputKey>
            <InputKey value=\\"PARENTHESES\\" label=\\"( )\\" type=\\"util\\">
              <_default type=\\"util\\" label=\\"( )\\" value=\\"PARENTHESES\\" onClick={[Function: onClick]}>
                <styled.div>
                  <div className=\\"sc-AxhCb jdovTQ\\">
                    <styled.div>
                      <div className=\\"sc-AxirZ eFNAXz\\">
                        <styled.div type=\\"util\\" label=\\"( )\\" value=\\"PARENTHESES\\" onClick={[Function: onClick]}>
                          <div type=\\"util\\" label=\\"( )\\" value=\\"PARENTHESES\\" onClick={[Function: onClick]} className=\\"sc-AxiKw hROJJW\\">
                            ( )
                          </div>
                        </styled.div>
                      </div>
                    </styled.div>
                  </div>
                </styled.div>
              </_default>
            </InputKey>
            <InputKey value=\\" ÷ \\" label=\\" ÷ \\" type=\\"operator\\">
              <_default type=\\"operator\\" label=\\" ÷ \\" value=\\" ÷ \\" onClick={[Function: onClick]}>
                <styled.div>
                  <div className=\\"sc-AxhCb jdovTQ\\">
                    <styled.div>
                      <div className=\\"sc-AxirZ eFNAXz\\">
                        <styled.div type=\\"operator\\" label=\\" ÷ \\" value=\\" ÷ \\" onClick={[Function: onClick]}>
                          <div type=\\"operator\\" label=\\" ÷ \\" value=\\" ÷ \\" onClick={[Function: onClick]} className=\\"sc-AxiKw ezXvin\\">
                             ÷ 
                          </div>
                        </styled.div>
                      </div>
                    </styled.div>
                  </div>
                </styled.div>
              </_default>
            </InputKey>
            <InputKey value={7} label=\\"7\\" type=\\"num\\">
              <_default type=\\"num\\" label=\\"7\\" value={7} onClick={[Function: onClick]}>
                <styled.div>
                  <div className=\\"sc-AxhCb jdovTQ\\">
                    <styled.div>
                      <div className=\\"sc-AxirZ eFNAXz\\">
                        <styled.div type=\\"num\\" label=\\"7\\" value={7} onClick={[Function: onClick]}>
                          <div type=\\"num\\" label=\\"7\\" value={7} onClick={[Function: onClick]} className=\\"sc-AxiKw dwulIh\\">
                            7
                          </div>
                        </styled.div>
                      </div>
                    </styled.div>
                  </div>
                </styled.div>
              </_default>
            </InputKey>
            <InputKey value={8} label=\\"8\\" type=\\"num\\">
              <_default type=\\"num\\" label=\\"8\\" value={8} onClick={[Function: onClick]}>
                <styled.div>
                  <div className=\\"sc-AxhCb jdovTQ\\">
                    <styled.div>
                      <div className=\\"sc-AxirZ eFNAXz\\">
                        <styled.div type=\\"num\\" label=\\"8\\" value={8} onClick={[Function: onClick]}>
                          <div type=\\"num\\" label=\\"8\\" value={8} onClick={[Function: onClick]} className=\\"sc-AxiKw dwulIh\\">
                            8
                          </div>
                        </styled.div>
                      </div>
                    </styled.div>
                  </div>
                </styled.div>
              </_default>
            </InputKey>
            <InputKey value={9} label=\\"9\\" type=\\"num\\">
              <_default type=\\"num\\" label=\\"9\\" value={9} onClick={[Function: onClick]}>
                <styled.div>
                  <div className=\\"sc-AxhCb jdovTQ\\">
                    <styled.div>
                      <div className=\\"sc-AxirZ eFNAXz\\">
                        <styled.div type=\\"num\\" label=\\"9\\" value={9} onClick={[Function: onClick]}>
                          <div type=\\"num\\" label=\\"9\\" value={9} onClick={[Function: onClick]} className=\\"sc-AxiKw dwulIh\\">
                            9
                          </div>
                        </styled.div>
                      </div>
                    </styled.div>
                  </div>
                </styled.div>
              </_default>
            </InputKey>
            <InputKey value=\\" × \\" label=\\" × \\" type=\\"operator\\">
              <_default type=\\"operator\\" label=\\" × \\" value=\\" × \\" onClick={[Function: onClick]}>
                <styled.div>
                  <div className=\\"sc-AxhCb jdovTQ\\">
                    <styled.div>
                      <div className=\\"sc-AxirZ eFNAXz\\">
                        <styled.div type=\\"operator\\" label=\\" × \\" value=\\" × \\" onClick={[Function: onClick]}>
                          <div type=\\"operator\\" label=\\" × \\" value=\\" × \\" onClick={[Function: onClick]} className=\\"sc-AxiKw ezXvin\\">
                             × 
                          </div>
                        </styled.div>
                      </div>
                    </styled.div>
                  </div>
                </styled.div>
              </_default>
            </InputKey>
            <InputKey value={4} label=\\"4\\" type=\\"num\\">
              <_default type=\\"num\\" label=\\"4\\" value={4} onClick={[Function: onClick]}>
                <styled.div>
                  <div className=\\"sc-AxhCb jdovTQ\\">
                    <styled.div>
                      <div className=\\"sc-AxirZ eFNAXz\\">
                        <styled.div type=\\"num\\" label=\\"4\\" value={4} onClick={[Function: onClick]}>
                          <div type=\\"num\\" label=\\"4\\" value={4} onClick={[Function: onClick]} className=\\"sc-AxiKw dwulIh\\">
                            4
                          </div>
                        </styled.div>
                      </div>
                    </styled.div>
                  </div>
                </styled.div>
              </_default>
            </InputKey>
            <InputKey value={5} label=\\"5\\" type=\\"num\\">
              <_default type=\\"num\\" label=\\"5\\" value={5} onClick={[Function: onClick]}>
                <styled.div>
                  <div className=\\"sc-AxhCb jdovTQ\\">
                    <styled.div>
                      <div className=\\"sc-AxirZ eFNAXz\\">
                        <styled.div type=\\"num\\" label=\\"5\\" value={5} onClick={[Function: onClick]}>
                          <div type=\\"num\\" label=\\"5\\" value={5} onClick={[Function: onClick]} className=\\"sc-AxiKw dwulIh\\">
                            5
                          </div>
                        </styled.div>
                      </div>
                    </styled.div>
                  </div>
                </styled.div>
              </_default>
            </InputKey>
            <InputKey value={6} label=\\"6\\" type=\\"num\\">
              <_default type=\\"num\\" label=\\"6\\" value={6} onClick={[Function: onClick]}>
                <styled.div>
                  <div className=\\"sc-AxhCb jdovTQ\\">
                    <styled.div>
                      <div className=\\"sc-AxirZ eFNAXz\\">
                        <styled.div type=\\"num\\" label=\\"6\\" value={6} onClick={[Function: onClick]}>
                          <div type=\\"num\\" label=\\"6\\" value={6} onClick={[Function: onClick]} className=\\"sc-AxiKw dwulIh\\">
                            6
                          </div>
                        </styled.div>
                      </div>
                    </styled.div>
                  </div>
                </styled.div>
              </_default>
            </InputKey>
            <InputKey value=\\" − \\" label=\\" − \\" type=\\"operator\\">
              <_default type=\\"operator\\" label=\\" − \\" value=\\" − \\" onClick={[Function: onClick]}>
                <styled.div>
                  <div className=\\"sc-AxhCb jdovTQ\\">
                    <styled.div>
                      <div className=\\"sc-AxirZ eFNAXz\\">
                        <styled.div type=\\"operator\\" label=\\" − \\" value=\\" − \\" onClick={[Function: onClick]}>
                          <div type=\\"operator\\" label=\\" − \\" value=\\" − \\" onClick={[Function: onClick]} className=\\"sc-AxiKw ezXvin\\">
                             − 
                          </div>
                        </styled.div>
                      </div>
                    </styled.div>
                  </div>
                </styled.div>
              </_default>
            </InputKey>
            <InputKey value={1} label=\\"1\\" type=\\"num\\">
              <_default type=\\"num\\" label=\\"1\\" value={1} onClick={[Function: onClick]}>
                <styled.div>
                  <div className=\\"sc-AxhCb jdovTQ\\">
                    <styled.div>
                      <div className=\\"sc-AxirZ eFNAXz\\">
                        <styled.div type=\\"num\\" label=\\"1\\" value={1} onClick={[Function: onClick]}>
                          <div type=\\"num\\" label=\\"1\\" value={1} onClick={[Function: onClick]} className=\\"sc-AxiKw dwulIh\\">
                            1
                          </div>
                        </styled.div>
                      </div>
                    </styled.div>
                  </div>
                </styled.div>
              </_default>
            </InputKey>
            <InputKey value={2} label=\\"2\\" type=\\"num\\">
              <_default type=\\"num\\" label=\\"2\\" value={2} onClick={[Function: onClick]}>
                <styled.div>
                  <div className=\\"sc-AxhCb jdovTQ\\">
                    <styled.div>
                      <div className=\\"sc-AxirZ eFNAXz\\">
                        <styled.div type=\\"num\\" label=\\"2\\" value={2} onClick={[Function: onClick]}>
                          <div type=\\"num\\" label=\\"2\\" value={2} onClick={[Function: onClick]} className=\\"sc-AxiKw dwulIh\\">
                            2
                          </div>
                        </styled.div>
                      </div>
                    </styled.div>
                  </div>
                </styled.div>
              </_default>
            </InputKey>
            <InputKey value={3} label=\\"3\\" type=\\"num\\">
              <_default type=\\"num\\" label=\\"3\\" value={3} onClick={[Function: onClick]}>
                <styled.div>
                  <div className=\\"sc-AxhCb jdovTQ\\">
                    <styled.div>
                      <div className=\\"sc-AxirZ eFNAXz\\">
                        <styled.div type=\\"num\\" label=\\"3\\" value={3} onClick={[Function: onClick]}>
                          <div type=\\"num\\" label=\\"3\\" value={3} onClick={[Function: onClick]} className=\\"sc-AxiKw dwulIh\\">
                            3
                          </div>
                        </styled.div>
                      </div>
                    </styled.div>
                  </div>
                </styled.div>
              </_default>
            </InputKey>
            <InputKey value=\\" + \\" label=\\" + \\" type=\\"operator\\">
              <_default type=\\"operator\\" label=\\" + \\" value=\\" + \\" onClick={[Function: onClick]}>
                <styled.div>
                  <div className=\\"sc-AxhCb jdovTQ\\">
                    <styled.div>
                      <div className=\\"sc-AxirZ eFNAXz\\">
                        <styled.div type=\\"operator\\" label=\\" + \\" value=\\" + \\" onClick={[Function: onClick]}>
                          <div type=\\"operator\\" label=\\" + \\" value=\\" + \\" onClick={[Function: onClick]} className=\\"sc-AxiKw ezXvin\\">
                             + 
                          </div>
                        </styled.div>
                      </div>
                    </styled.div>
                  </div>
                </styled.div>
              </_default>
            </InputKey>
            <InputKey value=\\"NEGATIVE\\" label={{...}} type=\\"util\\">
              <_default type=\\"util\\" label={{...}} value=\\"NEGATIVE\\" onClick={[Function: onClick]}>
                <styled.div>
                  <div className=\\"sc-AxhCb jdovTQ\\">
                    <styled.div>
                      <div className=\\"sc-AxirZ eFNAXz\\">
                        <styled.div type=\\"util\\" label={{...}} value=\\"NEGATIVE\\" onClick={[Function: onClick]}>
                          <div type=\\"util\\" label={{...}} value=\\"NEGATIVE\\" onClick={[Function: onClick]} className=\\"sc-AxiKw jUqkXN\\">
                            <div>
                              <strong>
                                +
                              </strong>
                              /
                              <strong>
                                −
                              </strong>
                            </div>
                          </div>
                        </styled.div>
                      </div>
                    </styled.div>
                  </div>
                </styled.div>
              </_default>
            </InputKey>
            <InputKey value={0} label=\\"0\\" type=\\"num\\">
              <_default type=\\"num\\" label=\\"0\\" value={0} onClick={[Function: onClick]}>
                <styled.div>
                  <div className=\\"sc-AxhCb jdovTQ\\">
                    <styled.div>
                      <div className=\\"sc-AxirZ eFNAXz\\">
                        <styled.div type=\\"num\\" label=\\"0\\" value={0} onClick={[Function: onClick]}>
                          <div type=\\"num\\" label=\\"0\\" value={0} onClick={[Function: onClick]} className=\\"sc-AxiKw dwulIh\\">
                            0
                          </div>
                        </styled.div>
                      </div>
                    </styled.div>
                  </div>
                </styled.div>
              </_default>
            </InputKey>
            <InputKey value=\\"DECIMAL\\" label=\\".\\" type=\\"util\\">
              <_default type=\\"util\\" label=\\".\\" value=\\"DECIMAL\\" onClick={[Function: onClick]}>
                <styled.div>
                  <div className=\\"sc-AxhCb jdovTQ\\">
                    <styled.div>
                      <div className=\\"sc-AxirZ eFNAXz\\">
                        <styled.div type=\\"util\\" label=\\".\\" value=\\"DECIMAL\\" onClick={[Function: onClick]}>
                          <div type=\\"util\\" label=\\".\\" value=\\"DECIMAL\\" onClick={[Function: onClick]} className=\\"sc-AxiKw fDiuPP\\">
                            .
                          </div>
                        </styled.div>
                      </div>
                    </styled.div>
                  </div>
                </styled.div>
              </_default>
            </InputKey>
            <InputKey value=\\"EQUALS\\" label=\\"=\\" type=\\"equals\\">
              <_default type=\\"equals\\" label=\\"=\\" value=\\"EQUALS\\" onClick={[Function: onClick]}>
                <styled.div>
                  <div className=\\"sc-AxhCb jdovTQ\\">
                    <styled.div>
                      <div className=\\"sc-AxirZ eFNAXz\\">
                        <styled.div type=\\"equals\\" label=\\"=\\" value=\\"EQUALS\\" onClick={[Function: onClick]}>
                          <div type=\\"equals\\" label=\\"=\\" value=\\"EQUALS\\" onClick={[Function: onClick]} className=\\"sc-AxiKw cuUIap\\">
                            =
                          </div>
                        </styled.div>
                      </div>
                    </styled.div>
                  </div>
                </styled.div>
              </_default>
            </InputKey>
          </div>
        </styled.div>
      </KeyPad>
    </div>
  </styled.div>
</Calculator>"
`;
